### PR TITLE
vCenter-Content Enrichment

### PR DESCRIPTION
**Pull request by rushriva**
Required items, please complete
   
   Change(s):
   - Workbook, Analytical rules added, Parser -updated

   Reason for Change(s):
   - Content enrichment of vCenter solution based on customer feedback

   Version Updated:
   - Yes
 
   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-mchatla** [comment]
Hi @rushriva 
Can you please resolve the failed checks.
Thanks

**Comment by NikTripathi** [comment]
@aprakash13 Can you please review and approve analytic rules?
Thanks

File: Solutions/VMware vCenter/Analytic Rules/vCenter-Root impersonation.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1078
+query: |
+ vCenter
+  | where Message has_all ('ImpersonateUser', 'VcIntegrity', 'root')
+  | extend user = 'root'
+  | extend AccountCustomEntity = user
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenter-Root impersonation.yaml:22]: Do not use custom entity mapping, it is legacy and can cause issues with the rules wizard.

File: Solutions/VMware vCenter/Analytic Rules/vCenter-Root impersonation.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: AccountCustomEntity
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenter-Root impersonation.yaml:27]: Change to just `user` field.

File: Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  let t_lookback = 1h;
+  let root_ips = vCenter
+  | where TimeGenerated between (ago(p_lookback) .. ago(t_lookback))
+  | where EventType has_all ('UserLoginSessionEvent', 'root', 'logged in')
+  | summarize make_set(SourceIP,128);
+  vCenter
+  | where TimeGenerated > ago(t_lookback)
+  | where EventType   has_all ('UserLoginSessionEvent', 'root', 'logged in')
+  | where SourceIP   !in (root_ips)
+  | extend IPCustomEntity = SourceIP
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml:30]: Do not use custom entity mapping, it is legacy and can cause issues with the rules wizard.

File: Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPCustomEntity
+version: 1.0.1
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml:36]: This is new, should be version 1.0.0.

File: Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  vCenter
+  | where TimeGenerated > ago(t_lookback)
+  | where EventType   has_all ('UserLoginSessionEvent', 'root', 'logged in')
+  | where SourceIP   !in (root_ips)
+  | extend IPCustomEntity = SourceIP
+entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPCustomEntity
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml:35]: Change to field `SourceIP`

File: Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+id: 03e8a895-b5ba-49a0-aed3-f9a997d92fbe
+name: vCenter - Unusual login
```
Comment (shainw) [inline Solutions/VMware vCenter/Analytic Rules/vCenterUnusualLogin.yaml:2]: The name doesn't seem to match the description.

**Review by shainw — CHANGES_REQUESTED** [review]

**Review by shainw — DISMISSED** [review]
Analytic Rules approved.

File: Solutions/VMware vCenter/Parsers/vCenter.txt [hunk 7]
```diff
@@ -22,63 +22,21 @@
...
+| where Message has ("UserLoginSessionEvent")
+| parse Message  with * "Event [" EventId:string "] [1-1] [" EventTime:datetime "] [" EventType:string "] [" EventSeverity "]" * "[User " DomainName:string "\\" Username:string "@" ClientIP " logged in as " UserAgent:string "]"
+};
+let vCenter_Logout =() {vcenter_CL
+| where Message has ("UserLogoutSessionEvent")
+| parse Message with * "Event [" EventId:string "] [1-1] [" EventTime:datetime "] [" EventType:string "] [" EventSeverity "]" * "[User" DomainName:string "\\" Username:string "@" ClientIP " logged out (login time:" LoginTime:string ", number of API invocations: " APIInvocationCount:int ", user agent:" UserAgent:string ")]"
+};
+let vCenter_Role=() {vcenter_CL
+| where Message has_any("RoleAddedEvent","RoleRemovedEvent")
+| parse Message with eventdate:string " " eventtime:string " " Hostname:string "Event [" EventId:string "] [1-1] [" EventTime:datetime "] [" EventType:string "] [" EventSeverity:string "]" * "[New role " RoleName:string " " Operation:string  "]"
```
Comment (devikamehra) [inline Solutions/VMware vCenter/Parsers/vCenter.txt:36]: Based on line 40, vpxd will also be present in this event. If we have to remove it from Hostname, it should be done here as well.

Comment (devikamehra) [inline Solutions/VMware vCenter/Parsers/vCenter.txt:36]: Is "eventtime" different than EventTime?

**Review by devikamehra — CHANGES_REQUESTED** [review]
Two small clarifications. Rest looks good.

File: Solutions/VMware vCenter/Parsers/vCenter.txt (hunk 7 above)
Comment (rushriva) [reply Solutions/VMware vCenter/Parsers/vCenter.txt]: Fixed

Comment (rushriva) [reply Solutions/VMware vCenter/Parsers/vCenter.txt]: Not needed, removed

**Review by devikamehra — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**